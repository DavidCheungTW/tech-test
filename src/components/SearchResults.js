import React from "react";
import "../styles/search-results.css";

const SearchResults = ({ results }) => {
  return (
    results &&
    results.length > 0 && (
      <p className="results">
        {results.map((image, index) => (
          <img
            key={index}
            className="image"
            src={image.href}
            title={image.title}
          />
        ))}
      </p>
    )
  );
};

export default SearchResults;
