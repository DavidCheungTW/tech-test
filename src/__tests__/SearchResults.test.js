import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";
import renderer from "react-test-renderer";

describe("SearchResults", () => {
  const validProps = ["image-href1", "image-href2", "image-href3"];

  test("Renders as expected", () => {
    const rendered = renderer.create(<SearchResults results={validProps} />);

    expect(rendered).toMatchSnapshot();
  });

  test("Assert 3 images are present", () => {
    render(<SearchResults results={validProps} />);

    const imgs = screen.getAllByRole("img");
    expect(imgs).toHaveLength(3);
  });
});
