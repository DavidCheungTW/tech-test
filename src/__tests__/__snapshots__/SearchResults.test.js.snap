// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults Renders as expected 1`] = `
<p
  className="results"
>
  <img
    className="image"
  />
  <img
    className="image"
  />
  <img
    className="image"
  />
</p>
`;
